src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs

```diff
@@ -279,7 +273,7 @@ public static class DependencyGraphRestoreUtility
             cacheContext.SolutionSpec = solutionDgSpec;
             cacheContext.SolutionSpecHash = newDependencyGraphSpecHash;
 
-            if (oldDependencyGraphSpecHash != newDependencyGraphSpecHash)
+            if (forceRestore || (oldDependencyGraphSpecHash != newDependencyGraphSpecHash))
             {
```

Comment:
It would. But - 
1. We are not doing a full restore. Just populating cache for the next restore comparison.
2. Having a fast restore, but adding time to next restore is also not correct.